mcs/class/corlib/System.Reflection/FieldInfo.cs

```diff
@@ -280,6 +283,12 @@ internal CustomAttributeData[] GetPseudoCustomAttributesData ()
 			return types;
 		}
 
+		public override int MetadataToken {
```

Comment:
This is probably wrong.

FieldInfo is abstract and its implementation probably throws instead. 
This should be on MonoField. 
You should add a test that verify that this overriding is correct.